vumi/transports/smpp/smpp_transport.py

```diff
@@ -306,36 +306,40 @@ def on_smpp_bind(self):
         yield self.publish_status_bound()
 
     def publish_status_unbound(self):
-        return self.publish_status('smpp', 'major', reasons=[{
-            'type': 'unbound',
-            'message': 'Connected but not bound',
-        }])
+        return self.publish_status(
+            status='major',
+            component='smpp',
+            type='unbound',
+            message='Connected but not bound')
 
     def publish_status_bound(self):
-        return self.publish_status('smpp', 'good', reasons=[{
-            'type': 'bound',
-            'message': 'Bound',
-        }])
+        return self.publish_status(
+            status='good',
+            component='smpp',
+            type='bound',
+            message='Bound')
 
     @inlineCallbacks
     def on_smpp_bind_timeout(self):
         yield self.publish_status_bind_timeout()
 
     def publish_status_bind_timeout(self):
-        return self.publish_status('smpp', 'major', reasons=[{
-            'type': 'bind_timeout',
-            'message': 'Timed out awaiting bind',
-        }])
+        return self.publish_status(
+            status='major',
+            component='smpp',
+            type='bind_timeout',
+            message='Timed out awaiting bind')
 
     @inlineCallbacks
     def on_connection_lost(self, reason):
         yield self.publish_status_connection_lost(reason)
 
     def publish_status_connection_lost(self, reason):
-        return self.publish_status('smpp', 'major', reasons=[{
-            'type': 'connection_lost',
-            'message': str(reason.value),
-        }])
+        return self.publish_status(
+            status='major',
+            component='smpp',
+            type='connection_lost',
+            message=str(reason.value))
```

Comment:
Wondering if this should be `unicode(reason.value)` since this all gets UTF-8 encoded by the JSON encoder later? Perhaps not something for this PR.